src/Configuration.cpp

```diff
@@ -333,25 +311,15 @@ void Configuration::parseAudio(void* _n)
  *
  * \note	Use of void pointer in declaration to avoid implementation details in header.
  */
-void Configuration::parseOptions(void* _n)
+void Configuration::parseOptions(XmlElement* element)
 {
-	// NOTE: Void pointer used to avoid implementation details in the class declaration.
-	XmlElement* element = static_cast<XmlNode*>(_n)->toElement();
-
-	// Probably not a necessary check but here for robustness.
-	if (!element)
-	{
-		std::cout << "Unexpected XML tag '" << static_cast<XmlNode*>(_n)->value() << "' found in configuration file while processing the '<graphics>' element." << std::endl;
-		return;
-	}
-
-	for (auto xmlNode = element->iterateChildren(nullptr);
-		 xmlNode != nullptr;
-		 xmlNode = element->iterateChildren(xmlNode))
+	for (auto setting = element->firstChildElement();
+		 setting != nullptr;
+		 setting = setting->nextSiblingElement())
```

Comment:
That reminds me, need to add ForEachElement/Attribute functions. This type of code is just gross. :P

Reply:
I know, right. It would be so much nicer.